include/dlaf/mpi_header.h

```diff
@@ -0,0 +1,43 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#if __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+#pragma GCC diagnostic ignored "-Wsign-conversion"
```

Comment:
I don't know if `-isystem` is the right option for this problem, moreover here it can not be applied easily.

In CMake the flag `-isystem` can be enabled in the directive `target_include_directories` with the option `SYSTEM`; Using FindMPI, which is highly suggested, we are not directly managing MPI include directories and so we cannot specify the SYSTEM option, since we only get a CMake target already populated.

It is possible to argue with this decision, but I think this is due to the fact that the `SYSTEM` option is just a workaround for the problem and not the right solution.

Reply:
Anyway warnings are not set yet. This should be addressed in another PR.